src/main/java/corgi/commands/MarkTaskCommand.java

```diff
@@ -0,0 +1,63 @@
+package corgi.commands;
+
+import corgi.storage.Storage;
+import corgi.tasks.Task;
+import corgi.tasks.TaskList;
+import corgi.tasks.TaskListIndexOutOfBoundsException;
+import corgi.tasks.TaskStatusException;
+import corgi.ui.Ui;
+
+/**
+ * Represents a command to mark a task as done or undone in the task list.
+ * This command updates the status of a task at the specified index in the task list.
+ */
+public class MarkTaskCommand extends Command {
+    /**
+     * The index of the task to be marked.
+     */
+    private int index;
+
+    /**
+     * The new status of the task (true for done, false for undone).
+     */
+    private boolean status;
```

Comment:
Would it be better to rename this variable as "isDone" instead?

Reply:
Thanks for the suggestion, will consider it.